From 861b6c6a4a437579ee5fd8e617bf67d5409f699f Mon Sep 17 00:00:00 2001
From: Ankit Jain <ankitja@vmware.com>
Date: Fri, 5 Aug 2022 05:58:25 +0000
Subject: [PATCH 05/14] isoInstaller: Refresh devices in retries if mount fails

- Device not detected properly and mount fails
- Fixed by adding refresh_devices() in retries

Change-Id: Ia7cdfa041b2d7a3a9fc9685166e3c19d49a1240e
Signed-off-by: Ankit Jain <ankitja@vmware.com>
---
 photon_installer/isoInstaller.py | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/photon_installer/isoInstaller.py b/photon_installer/isoInstaller.py
index 799d29a..b369f26 100755
--- a/photon_installer/isoInstaller.py
+++ b/photon_installer/isoInstaller.py
@@ -16,6 +16,7 @@ from argparse import ArgumentParser
 from installer import Installer
 from commandutils import CommandUtils
 from jsonwrapper import JsonWrapper
+from device import Device
 
 class IsoInstaller(object):
     def __init__(self, options):
@@ -163,8 +164,9 @@ class IsoInstaller(object):
             retval = process.wait()
             if retval == 0:
                 return mount_path
-            print("Failed to mount the device, retry in a second")
-            time.sleep(1)
+            print("Failed to mount the device, retry in 5 seconds")
+            Device.refresh_devices()
+            time.sleep(5)
         print("Failed to mount the device, exiting the installer")
         print("check the logs for more details")
         raise Exception("Can not mount the device")
-- 
2.23.3

